src/Microsoft.Kusto.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -92,32 +92,29 @@ public async Task HandleScriptExecuteRequest(ScriptingParams parameters, Request
                 // if a connection string wasn't provided as a parameter then
                 // use the owner uri property to lookup its associated ConnectionInfo
                 // and then build a connection string out of that
-                if (parameters.ConnectionString == null)
+                if (parameters.DatabaseName == null)
                 {
                     ConnectionInfo connInfo;
                     _connectionService.TryFindConnection(parameters.OwnerUri, out connInfo);
                     if (connInfo != null)
                     {
-                        parameters.ConnectionString = ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
+                        parameters.DatabaseName = connInfo.ConnectionDetails.DatabaseName;
                     }
                     else
                     {
                         throw new Exception("Could not find ConnectionInfo");
                     }
                 }
 
-                SmoScriptingOperation operation;
-                var datasource = _connectionService.GetOrOpenConnection(parameters.OwnerUri, ConnectionType.Default)
-                    .Result.GetUnderlyingConnection();
                 if (!ShouldCreateScriptAsOperation(parameters))
                 {
-                    operation = new ScriptingScriptOperation(parameters, datasource);
-                }
-                else
-                {
-                    operation = new ScriptAsScriptingOperation(parameters, _scripter, datasource);
+                    throw new InvalidOperationException("Unable to create script.");
                 }
+                
+                var datasource = _connectionService.GetOrOpenConnection(parameters.OwnerUri, ConnectionType.Default)
+                    .Result.GetUnderlyingConnection();
 
+                SmoScriptingOperation operation = new ScriptAsScriptingOperation(parameters, _scripter, datasource);
```

Comment:
Not for this PR, but we should remove any Smo related code in the Kusto.ServiceLayer.